source/android/adaptivecards/src/main/java/com/microsoft/adaptivecards/renderer/action/HttpActionRenderer.java

```diff
@@ -0,0 +1,183 @@
+                        }
+                    }
+
+                    String pattern = "\\{\\{(" + TextUtils.join("|", data.keySet()) + ")\\.Value\\}\\}";
+                    urlDataBinded = replaceWithBindingData(m_httpAction.GetUrl(), pattern, data);
+                    bodyDataBinded = replaceWithBindingData(m_httpAction.GetBody(), pattern, data);
+                }
+
+                if (TextUtils.isEmpty(urlDataBinded))
+                {
+                    return new HttpRequestResult<byte[]>(new IllegalArgumentException("URL cannot be empty."));
+                }
+
+                if (TextUtils.isEmpty(bodyDataBinded))
+                {
+                    return new HttpRequestResult<byte[]>(new IllegalArgumentException("body cannot be empty."));
+                }
+
+                String method = m_httpAction.GetMethod().toUpperCase();
+                Map<String, String> requestProperty = null;
+                if (method.equals(HttpRequestHelper.HTTP_METHOD_POST))
+                {
+                    return new HttpRequestResult<byte[]>(HttpRequestHelper.post(urlDataBinded, requestProperty, bodyDataBinded));
+                }
+                else if (method.equals(HttpRequestHelper.HTTP_METHOD_GET))
+                {
+                    return new HttpRequestResult<byte[]>(HttpRequestHelper.get(urlDataBinded, requestProperty));
+                }
+                else if (method.equals(HttpRequestHelper.HTTP_METHOD_PUT))
+                {
+                    return new HttpRequestResult<byte[]>(HttpRequestHelper.put(urlDataBinded, requestProperty, bodyDataBinded));
+                }
+                else
+                {
+                    throw new IllegalArgumentException("Unsupported HTTP method: " + m_httpAction.GetMethod());
+                }
+            }
+            catch (Exception excep)
+            {
+                return new HttpRequestResult<byte[]>(excep);
+            }
+        }
+
+        @Override
+        protected void onPostExecute(HttpRequestResult result)
+        {
+            super.onPostExecute(result);
+        }
+
+        private HttpAction m_httpAction;
+        private Vector<IInputHandler> m_inputHandlerList;
+    }
+
+    @Override
+    public Button render(
+            Context context,
+            ViewGroup viewGroup,
+            BaseActionElement baseActionElement,
+            Vector<IInputHandler> inputHandlerList,
+            IShowCardActionHandler showCardActionHandler,
+            ISubmitActionHandler submitActionHandler,
+            HostConfig hostConfig)
+    {
+        HttpAction httpAction = null;
+        if (baseActionElement instanceof HttpAction)
+        {
+            httpAction = (HttpAction) baseActionElement;
+        }
+        else if ((httpAction = HttpAction.dynamic_cast(baseActionElement)) == null)
+        {
+            throw new InternalError("Unable to convert BaseActionElement to ShowCardAction object model.");
```

Comment:
Good catch again. :)

---
In reply to: [126771606](https://github.com/Microsoft/AdaptiveCards/pull/461#discussion_r126771606) [](ancestors = 126771606)